scripts/build.py

```diff
@@ -549,6 +525,38 @@ def save_hashes_as_json(target_filepath, file_hashes):
     file_hashes[relative_filepath] = file_hash
 
 
+def minify_func(source_path, target_path, file_hashes, filename):
+    if filename.endswith('.html'):
+        process_html(source_path, target_path, file_hashes)
+    elif filename.endswith('.css'):
+        _minify(source_path, target_path)
+    elif filename.endswith('.js'):
+        _minify(source_path, target_path)
+    else:
+        shutil.copyfile(source_path, target_path)
+
+
+def _execute_tasks(tasks, batch_size=24):
+    """Starts all tasks and checks the results.
+
+    Runs no more than 'batch_size' tasks at a time.
+    """
+    remaining_tasks = [] + tasks
+    currently_running_tasks = set([])
+
+    while remaining_tasks or currently_running_tasks:
+        if currently_running_tasks:
+            for task in list(currently_running_tasks):
+                task.join(1)
+                if not task.isAlive():
+                    currently_running_tasks.remove(task)
```

Comment:
This won't be necessary if you delete from a list using an index. Also, hashing the `task` does slow things down a bit.

Reply:
Done.

Reply:
Refer to above comment about why I don't think this is the right move.